assets/js/components/post-searcher.js

```diff
@@ -112,28 +115,37 @@ class PostSearcher extends Component {
 	}
 
 	onConfirm( selection ) {
-		this.setState( {
-			selection,
-		} );
+		const { results } = this.state;
+		let match;
+		// Check to that the selection is "valid".
+		if ( ! Array.isArray( results ) || ! this.disallowedSelections.includes( selection ) ) {
+			match = results.find( ( result ) => result.post_title === selection );
+			if ( selection && ! isUndefined( match ) ) {
```

Comment:
This is a bit of a nitpick, but let's not use lodash for this. We've agreed to use a simple strict comparison with `undefined` so that's what we should use going forward for consistency. Sorry for not mentioning this before, but lodash does the same thing internally so it doesn't offer much benefit, other than semantics which are not bad, so this is just from a consistency standpoint.

In this case specifically, I think we can simply check `match` as a basic truthy check as it will either be a post `Object`, or `undefined` 🙂 
```suggestion
			if ( selection && match ) {
```

Reply:
Updated.